lib/collections.js

```diff
@@ -157,13 +157,6 @@ var factory = module.exports = function (request) {
 
             if (query && typeof query === 'object') {
```

Comment:
If this PR is correct and accepted, the only thing left in this condition is to overwrite `requestObject.endpoint`. If I'm correct, before it's overwritten here, it would be something like `pods/[object Object]`, and this would overwrite it to `pods`. I think this is a case that should be handled by the `getPath` function, since the `/[object Object]` part is likely never desirable.

I think making the following change on [line 11](https://github.com/tenxcloud/node-kubernetes-client/blob/master/lib/collections.js#L11) would achieve this:

```diff
- return each !== null && typeof each !== 'undefined';
+ return typeof each !== 'object' && typeof each !== 'undefined';
```

In the above replacement, the initial null check would no longer be needed, since `typeof null === 'object'`.

If you'd like, I can make that change in this PR (or in a separate one?)?